app/src/androidTest/java/org/mozilla/fenix/ui/SettingsHomepageTest.kt

```diff
@@ -208,6 +197,29 @@ class SettingsHomepageTest {
         }
     }
 
+    @Test
+    fun ignoreStartOnHomeWhenLaunchedByExternalLinkTest() {
+        val genericPage = getGenericAsset(mockWebServer, 1)
+
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openHomepageSubMenu {
+            clickStartOnHomepageButton()
+        }.goBack {}
+
+        with(activityIntentTestRule) {
+            finishActivity()
+            mDevice.waitForIdle()
+            openAppFromExternalLink(genericPage.url.toString())
+        }
+
+        browserScreen {
+            dismissCookieProtectionHint()
```

Comment:
@Mugurell Was there a way to disable TCP after activity launch, or was that previous crash still a problem? I'm just wondering as then we won't need to dismiss it here as part of a test.

Reply:
I made some changes and now I'm disabling TCP before re-launching the app. Please have another look

Reply:
The `applySettingsExceptions` method Oana used in the new patch version is the right approach to immediately modify these settings in individual tests while also ensuring their proper cleanup.